cns/restserver/util.go

```diff
@@ -40,6 +40,35 @@ func (service *HTTPRestService) setNetworkInfo(networkName string, networkInfo *
 	service.state.Networks[networkName] = networkInfo
 }
 
+func (service *HTTPRestService) SavePnpIDMacaddressMapping(ctx context.Context) error {
+	// If mapping is already set, skip setting it again.
+	if len(service.state.PnpIDByMacAddress) != 0 {
```

Comment:
I think there will be an issue with subsequent calls to this function, here is the timeline of events:
- `getPNPIDFromMacAddress` is called for a new pod, in which its MAC address doesnt exist in `service.state.PnpIDByMacAddress`
- `getPNPIDFromMacAddress` invokes `SavePnpIDMacaddressMapping`, but since `service.state.PnpIDByMacAddress` might be populated from the initial invoke, its length is > 0, so we return immediately, therefore not able to get the underlying updated PnpIDMacaddressMapping

Reply:
To give bit more context to this, Mapping doesn't change, it will remain constant. We are doing this in two places just to cover the case of failures during the start up.